### ISSUE TITLE
Procedural Texture asset component error

### ISSUE THREAD

**Issue by RueShejn** [Yellow-Dog-Man/Resonite-Issues#2410]
### Describe the bug?

when adding any of the procedural texture components they report an error at the bottom

### To Reproduce

make an empty object and attach any of the default procedural texture components

### Expected behavior

no error

### Screenshots

![Resonite_T7hJpgnTAK](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/65612998/a953a255-3e53-4f20-b124-e5089766d1e2)

### Resonite Version Number

Beta 2024.6.25.1149

### What Platforms does this occur on?

Windows

### What headset if any do you use?

quest pro, desktop

### Log Files

[DESKTOP-L6KVB13 - 2024.6.25.1149 - 2024-06-26 20_11_37.log](https://github.com/user-attachments/files/16006476/DESKTOP-L6KVB13.-.2024.6.25.1149.-.2024-06-26.20_11_37.log)

**Comment by epicEaston197** [comment]
That's probably not an error it's probably some weird Localization thing

**Comment by shiftyscales** [comment]
Seeking input from @BlueCyro - could you look into this component and see what locale key it is supposed to be displaying, and also check against the locale file and ensure that key exists?

Relevant exception from attaching this component is:

```
9:59:25 AM.542 (296 FPS)	Exception formatting message Inspector.ProceduralAsset.Error, Locale Code: en:
System.NullReferenceException: Object reference not set to an instance of an object
  at Jeffijoe.MessageFormat.MessageFormatter.FormatMessage (System.String pattern, System.Collections.Generic.IReadOnlyDictionary`2[TKey,TValue] args, System.Boolean ignoreMissingVariables) [0x0003b] in <ce50cea1308a4384b872fbd8c8ce5938>:0 
  at Elements.Assets.LocaleResource.Format (System.String key, System.Collections.Generic.Dictionary`2[TKey,TValue] arguments) [0x00021] in <8c482d83338649beb514aec144689af2>:0 
  at FrooxEngine.LocaleResource.Format (System.String key, System.Collections.Generic.Dictionary`2[TKey,TValue] arguments, System.Boolean returnNullIfNotFound) [0x0001b] in <0dd2210b70d74ab9a16b9780b914dbb9>:0 

  at System.Environment.get_StackTrace () [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at Elements.Core.UniLog.Error (System.String message, System.Boolean stackTrace) [0x00000] in <74f2d74bce2b4e51a46e4651f5b3c81c>:0 
  at FrooxEngine.LocaleResource.Format (System.String key, System.Collections.Generic.Dictionary`2[TKey,TValue] arguments, System.Boolean returnNullIfNotFound) [0x00000] in <0dd2210b70d74ab9a16b9780b914dbb9>:0 
  at FrooxEngine.LocaleHelper.SetLocalized (FrooxEngine.IField`1[T] field, System.String key, System.String format, System.Collections.Generic.Dictionary`2[TKey,TValue] arguments) [0x00000] in <0dd2210b70d74ab9a16b9780b914dbb9>:0 
  at FrooxEngine.LocaleHelper.AssignLocaleString (FrooxEngine.IField`1[T] field, Elements.Core.LocaleString localeString) [0x00000] in <0dd2210b70d74ab9a16b9780b914dbb9>:0 
  at FrooxEngine.UIX.Text.set_LocaleContent (Elements.Core.LocaleString value) [0x00000] in <0dd2210b70d74ab9a16b9780b914dbb9>:0 
  at FrooxEngine.UIX.UIBuilder.Text (Elements.Core.LocaleString& text, System.Boolean bestFit, System.Nullable`1[T] alignment, System.Boolean parseRTF, System.String nullContent) [0x00000] in <0dd2210b70d74ab9a16b9780b914dbb9>:0 
  at FrooxEngine.ProceduralTextureBase.BuildInspectorUI (FrooxEngine.UIX.UIBuilder ui) [0x00000] in <0dd2210b70d74ab9a16b9780b914dbb9>:0 
  at FrooxEngine.WorkerInspector.BuildUIForComponent (FrooxEngine.Worker worker, System.Boolean allowRemove, System.Boolean allowDuplicate, System.Predicate`1[T] memberFilter) [0x00000] in <0dd2210b70d74ab9a16b9780b914dbb9>:0 
  at FrooxEngine.WorkerInspector+<>c__DisplayClass24_0.<WorkerAdded>b__0 () [0x00000] in <0dd2210b70d74ab9a16b9780b914dbb9>:0 
  at FrooxEngine.World.RunSynchronousActions () [0x00000] in <0dd2210b70d74ab9a16b9780b914dbb9>:0 
  at FrooxEngine.World.NextRefreshStage () [0x00000] in <0dd2210b70d74ab9a16b9780b914dbb9>:0 
  at FrooxEngine.World.Refresh () [0x00000] in <0dd2210b70d74ab9a16b9780b914dbb9>:0 
  at FrooxEngine.WorldManager.UpdateStep () [0x00000] in <0dd2210b70d74ab9a16b9780b914dbb9>:0 
  at FrooxEngine.WorldManager.RunUpdateLoop () [0x00000] in <0dd2210b70d74ab9a16b9780b914dbb9>:0 
  at FrooxEngine.Engine.UpdateStep () [0x00000] in <0dd2210b70d74ab9a16b9780b914dbb9>:0 
  at FrooxEngine.Engine.RunUpdateLoop () [0x00000] in <0dd2210b70d74ab9a16b9780b914dbb9>:0 
  at UnityFrooxEngineRunner.FrooxEngineRunner.UpdateFrooxEngine () [0x00000] in <44daa1112e7a4a209c8f2223185e0c41>:0 
  at UnityFrooxEngineRunner.FrooxEngineRunner.Update () [0x00000] in <44daa1112e7a4a209c8f2223185e0c41>:0 
```

**Comment by Banane9** [comment]
Heh, that's an issue with it using the wrong overload of AsLocaleKey - using the (string, string, bool, Dictionary) one rather than the intended (string, string, object) one.

**Comment by BlueCyro** [comment]
Fix is waiting in internal PR :D

**Comment by XDelta** [comment]
Likely similar fix needed for https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/1430

**Comment by Banane9** [comment]
> Likely similar fix needed for #1430

Sure looks like it!

**Comment by Frooxius** [comment]
Merged in 2024.9.13.672! Thanks!
